server/app/views/CiviFormMarkdown.java

```diff
@@ -17,6 +19,37 @@ public String render(String markdown) {
     return RENDERER.render(markdownRootNode);
   }
 
+  /**
+   * Removes p tags that are added by default by the CommonMark library. Logic was pulled from this
+   * commit which is the recommended way to solve this problem:
+   * https://github.com/commonmark/commonmark-java/commit/8b0fd7c73afaf1756edb6412c15a75cc423a6ba9
+   */
```

Comment:
Please include details on why the `p` tags are getting removed

Reply:
done